### ISSUE TITLE
Implement a better error handler

### ISSUE THREAD

**Issue by tomsotte** [babel/gulp-babel#121]
I wanted to make a gulp watch task, but every time it stopped when an error were found.
Since I've searched but found no real workaround, the only solution I found was to handle the error like `gulp-sass` does. It does not interrupt the watch and just report it on the console.

Solution source: https://github.com/dlmanning/gulp-sass/blob/master/index.js#L178

An easy implementation for whoever has problem with it is:
```js
const babel = require('gulp-babel');

babel.logError = function logError(error) {
	gutil.log(error.toString(), '\n\b', error.codeFrame);
	this.emit('end');
};

// how to use
babel({...})
.on('error', babel.logError)
```

**Comment by n1ru4l** [comment]
Would be cool to have something similar to https://github.com/adametry/gulp-eslint

```javascript
  gulp.src(src)
    .pipe(babel())
    .pipe(babel.format())
```

**Comment by hereisnaman** [comment]
@tomsotte I guess giving the user, chance to work around error event is better than just logging and not throwing an error. 

User may need to do something when error event is fired. I am also doing it the same way you are while working with watchers.

**Comment by tomsotte** [comment]
@hereisnaman what I am proposing does not exclude yoru use case, just ease the logging of the error, which it seems to me is the most common case for the compilation of sass.

**Comment by demurgos** [comment]
I agree that we need a better error handler. It should not automatically log errors but simply give the user the ability to add its own custom handler.

Not stopping in watch-mode could be a specific option.

**Comment by tomsotte** [comment]
It is in fact what I'm proposing, a nifty addition that does not change in any way the current implementation.
It just simplifies and reduce code for the developer who just need to log errors while on watch.
